pkg/scheduler/framework/frameworkutils.go

```diff
@@ -70,3 +76,106 @@ func classifyBindings(policy *fleetv1beta1.ClusterSchedulingPolicySnapshot, bind
 
 	return bound, scheduled, obsolete, dangling
 }
+
+// crossReferencePickedCustersAndBindings cross references picked clusters in the current scheduling
+// run and existing bindings to find out:
+//
+//   - bindings that should be created, i.e., create a binding in the state of Scheduled for every
+//     cluster that is newly picked and does not have a binding associated with;
+//   - bindings that should be updated, i.e., associate a binding, whose target cluster is picked again
+//     in the current run, with the latest score and the latest scheduling policy snapshot (if applicable);
+//   - bindings that should be deleted, i.e., mark a binding as unschedulable if its target cluster is no
+//     longer picked in the current run.
+//
+// Note that this function will return bindings with all fields fulfilled/refreshed, as applicable.
+func crossReferencePickedCustersAndObsoleteBindings(
+	crpName string,
+	policy *fleetv1beta1.ClusterSchedulingPolicySnapshot,
+	picked ScoredClusters,
+	obsolete []*fleetv1beta1.ClusterResourceBinding,
+) (toCreate, toUpdate, toDelete []*fleetv1beta1.ClusterResourceBinding, err error) {
+	// Pre-allocate with a reasonable capacity.
+	toCreate = make([]*fleetv1beta1.ClusterResourceBinding, 0, len(picked))
+	toUpdate = make([]*fleetv1beta1.ClusterResourceBinding, 0, 20)
```

Comment:
curious why 20?

Reply:
Hi Ryan! It's just a number that's larger enough than the max. number of clusters in use by most interviewees as reported by the CNCF survey (10 IRC) but not too large to over-allocation memory.